astro-client/astro_test.go

```diff
@@ -896,3 +896,59 @@ func TestGetWorkspace(t *testing.T) {
 		assert.Error(t, err, "API error")
 	})
 }
+
+func TestWorkerQueueOptions(t *testing.T) {
+	testUtil.InitTestConfig(testUtil.CloudPlatform)
+	mockResponse := Response{
+		Data: ResponseData{
+			GetWorkerQueueOptions: WorkerQueueDefaultOptions{
+				MinWorkerCount: WorkerQueueOption{
+					Floor:   1,
+					Ceiling: 10,
+					Default: 2,
+				},
+				MaxWorkerCount: WorkerQueueOption{
+					Floor:   2,
+					Ceiling: 20,
+					Default: 5,
+				},
+				WorkerConcurrency: WorkerQueueOption{
+					Floor:   5,
+					Ceiling: 80,
+					Default: 25,
+				},
+			},
+		},
+	}
+	jsonResponse, err := json.Marshal(mockResponse)
+	assert.NoError(t, err)
+
+	t.Run("happy path", func(t *testing.T) {
+		client := testUtil.NewTestClient(func(req *http.Request) *http.Response {
+			return &http.Response{
+				StatusCode: 200,
+				Body:       io.NopCloser(bytes.NewBuffer(jsonResponse)),
+				Header:     make(http.Header),
+			}
+		})
+		astroClient := NewAstroClient(client)
+
+		resp, err := astroClient.GetWorkerQueueOptions()
+		assert.NoError(t, err)
+		assert.Equal(t, resp, mockResponse.Data.GetWorkerQueueOptions)
+	})
+
+	t.Run("error path", func(t *testing.T) {
```

Comment:
Are there other error paths that can occur when retrieving this info? If so, it might be worth renaming `error path` to something more descriptive.

Reply:
I doubt the comment. Can we describe what is the expected error path here, so its easier to understand what is covered and if some more cases are required

Reply:
@frank-ye @kushalmalani `GetWorkerQueueOptions()` is what the `astro.Client` calls to get the default worker queue options from astrohub API  via a graphql query.  The error path here is a catch all for any error returned by the public API call below:

```
	wqResp, err := wqReq.DoWithPublicClient(c)
	if err != nil {
		return WorkerQueueDefaultOptions{}, err
	}
``` 
I have made the description: `catch all for any API error` Is that better and more descriptive?

Reply:
I see, then i am good either way.